### ISSUE TITLE
where to change number of epochs and learning rate

### ISSUE THREAD

**Issue by vsbiradar** [OpenNMT/OpenNMT-py#603]
I want to change number of epochs the model should run and rate of decaying of learning rate. I am not able to find where the value opt.epochs is initialized to 13 and also decaying of learning rate.

**Comment by xforceco** [comment]
Running into this issue as well.
Current design is: 
`either epoch > start_decay_at or valid perplexity increases` => switch on decay forever

According to `update_learning_rate` in https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/Optim.py

Probably a better design to give user more control is needed
Wait for author's answer

**Comment by srush** [comment]
`-epochs` to set epochs
`-start_decay_at` will start decaying the rate at that epoch or if validation PPL decreases. 
`-learning_rate_decay` gives the rate of decay per epoch. 

We can add some other rates. if there is interest. We have a special `-decay_method` flag for the transformer.

**Comment by srush** [comment]
Code is pretty hackable, happens here 

https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/Optim.py#L131
